app/src/lib/git/commit.ts

```diff
@@ -46,20 +56,30 @@ export async function createCommit(
  */
 export async function createMergeCommit(
   repository: Repository,
-  files: ReadonlyArray<WorkingDirectoryFileChange>
+  files: ReadonlyArray<WorkingDirectoryFileChange>,
+  manualResolutions: ReadonlyMap<string, ManualConflictResolution> = new Map()
 ): Promise<string | undefined> {
   // Clear the staging area, our diffs reflect the difference between the
   // working directory and the last commit (if any) so our commits should
   // do the same thing.
   try {
     await unstageAll(repository)
-    await stageFiles(repository, files)
+
+    // apply manual conflict resolutions
+    for (const [path, resolution] of manualResolutions) {
+      const file = files.find(f => f.path === path)
+      if (file !== undefined) {
+        await stageManualConflictResolution(repository, file, resolution)
+      }
+    }
+
+    const otherFiles = files.filter(f => !manualResolutions.has(f.path))
+
+    await stageFiles(repository, otherFiles)
     const result = await git(
       [
-        'commit',
-        // no-edit here ensures the app does not accidentally invoke the user's editor
-        '--no-edit',
-        // By default Git merge commits do not contain any commentary (which
+        'commit', // no-edit here ensures the app does not accidentally invoke the user's editor
```

Comment:
:art: minor thing, but it looks like the comments got pulled up rather than being kept nearer to their relevant comments